src/main/java/rs/hybridit/serviceImpl/BookRentServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package rs.hybridit.serviceImpl;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import rs.hybridit.model.Book;
+import rs.hybridit.model.BookCopy;
+import rs.hybridit.model.Library;
+import rs.hybridit.model.User;
+import rs.hybridit.repository.BookCopyRepository;
+import rs.hybridit.repository.BookRepository;
+import rs.hybridit.repository.LibraryRepository;
+import rs.hybridit.repository.UserRepository;
+import rs.hybridit.service.BookRentService;
+
+@Service
+public class BookRentServiceImpl implements BookRentService {
+
+	private final BookCopyRepository bookCopyRepository;
+
+	private final BookRepository bookRepository;
+
+	private final LibraryRepository libraryRepository;
+
+	private final UserRepository userRepository;
+
+	public BookRentServiceImpl(BookCopyRepository bookCopyRepository, BookRepository bookRepository,
+		LibraryRepository libraryRepository, UserRepository userRepository) {
+		this.bookCopyRepository = bookCopyRepository;
+		this.bookRepository = bookRepository;
+		this.libraryRepository = libraryRepository;
+		this.userRepository = userRepository;
+	}
+
+	@Override
+	public BookCopy rentBookCopy(Long bookId) {
+		Book book = bookRepository.findById(bookId).orElse(null);
+		if (book != null) {
```

Comment:
Don't use optional like this but use map bookRepository.findById(bookId).map(book -> { ... }).orElseThrow(() -> new exception ...